### PR TITLE
Support deploying the controller

### PR DESCRIPTION
**Pull request by lmars**

### CODE REVIEW

File: controller/deployer/strategies/all-at-once.go [hunk 1]
```diff
@@ -6,67 +6,67 @@ func allAtOnce(d *Deploy) error {
...
 		}
-		expected[typ] = map[string]int{"down": n}
+		expected[typ] = map[string]int{"down": existing}
 	}
-	olog.Info("waiting for job events", "expected", expected)
-	if err := d.waitForJobEvents(d.OldReleaseID, expected, olog); err != nil {
-		olog.Error("error waiting for job events", "err", err)
-		return err
+	if expected.Count() > 0 {
+		olog := log.New("release_id", d.OldReleaseID)
```
Comment (titanous) [inline controller/deployer/strategies/all-at-once.go]: I'd just shadow `log` here and above (to avoid mistakes like the one made on line 59.

**Comment by lmars** [comment]
@titanous comments addressed

**Comment by lmars** [comment]
@titanous tests fixed (hopefully) by adding support for omni process types (i.e. the scheduler).

**Comment by lmars** [comment]
@titanous tests passing now, please review.

**Comment by titanous** [comment]
LGTM

**Pull request merged**